server/app/com/xsn/explorer/data/anorm/dao/BlockPostgresDAO.scala

```diff
@@ -345,22 +345,24 @@ class BlockPostgresDAO @Inject()(
       s"""
         |SELECT blk.*, COALESCE(tx.count, 0) transactions 
         |FROM (
-        |WITH CTE AS (
-        |  SELECT height as lastSeenHeight
-        |  FROM blocks
-        |  WHERE blockhash = {lastSeenHash}
-        |)
-        |SELECT blockhash, previous_blockhash, next_blockhash, merkle_root, height, time, difficulty
-        |FROM CTE CROSS JOIN blocks b
-        |WHERE b.height $comparator lastSeenHeight) blk
+        |   WITH CTE AS (
```

Comment:
Same, it will be clear with a test

Reply:
This was also for the same reason. Instead of joining all the items that fit the condition, we can only join the filtered items with the `height` and `limit`. The problem here is that the `GROUP BY` can change the order so even the item set is correct, the order can be wrong. That's why I added the `ORDER BY` clause at the end of the query.